application/backend/src/PublicHolidaysService/PublicHolidays/FrenchPublicHolidays.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\PublicHolidaysService\PublicHolidays;
+
+use App\Dictionary\PublicHolidayAvailableCountryDictionary;
+use App\PublicHolidaysService\AbstractPublicHolidays;
+
+class FrenchPublicHolidays extends AbstractPublicHolidays
+{
+    public function getCountry(): string
+    {
+        return PublicHolidayAvailableCountryDictionary::FRANCE;
+    }
+
+    public function getPublicHolidays(string $year, array $options = []): array
+    {
+        $joursFeries = [
+            $this->dimanchePaques((int)$year),
+            $this->lundiPaques((int)$year),
+            $this->jeudiAscension((int)$year),
+            $this->lundiPentecote((int)$year),
+            "$year-01-01",        // Nouvel an
+            "$year-05-01",        // Fête du travail
+            "$year-05-08",        // Armistice 1945
+            "$year-05-15",        // Assomption
+            "$year-07-14",        // Fête nationale
+            "$year-11-11",        // Armistice 1918
+            "$year-11-01",        // Toussaint
+            "$year-12-25"         // Noël
+        ];
+
+        if (isset($options['alsaceMoselle']) && $this->isTrue($options['alsaceMoselle'])) {
+            $joursFeries[] = "$year-12-26";
+            $joursFeries[] = $this->vendrediSaint((int)$year);
+        }
+
+        sort($joursFeries);
+
+        return $joursFeries;
+    }
+
+    private function dimanchePaques($annee)
+    {
+        return date("Y-m-d", easter_date($annee));
+    }
+
+    private function vendrediSaint($annee)
+    {
+        $dimanchePaques = $this->dimanchePaques($annee);
```

Comment:
I think you could pass the $dimanchePaque var by params, since you use it in every methods below. It would prevent instanciate many time the same dates. Wdyt ?

Reply:
Done